src/api/File.js

```diff
@@ -78,6 +82,44 @@ class File extends Item {
         this.successHandler(`${downloadBaseUrl}?${downloadUrlQuery}`);
     }
 
+    /**
+     * Polls a representation's infoUrl, attempting to generate a representation
+     *
+     * @param {FileRepresentation} representation - reprsentation that should have it's info.url polled
+     * @return {Promise<FileRepresentation>} - representation updated with most current status
+     */
+    async generateRepresentation(representation: ?FileRepresentation): Promise<?FileRepresentation> {
+        const infoUrl = getProp(representation, 'info.url');
+
+        const numOfTries = 3;
+        const initialTimeout = 3000;
+        const backoffFactor = 2;
+
+        if (!infoUrl) {
+            return representation;
+        }
+
+        const pollInfoUrl = async (resolve: Function, reject: Function): Promise<?string> => {
```

Comment:
This might be better served as a normal promise chain callback like:
```
    const isStatusSuccess = response => {
        const status = getProp(response, 'data.status.state');
        return !status || status === REPRESENTATIONS_RESPONSE_SUCCESS || status === REPRESENTATIONS_RESPONSE_ERROR);
    }
    return this.xhr.get({ url: infoUrl }).catch(() =>
        retryNumOfTimes(
            (successCallback, errorCallback) =>
                this.xhr
                    .get({ url: infoUrl })
                    .then(response => (isStatusSuccess(response) ? successCallback(response.data) : errorCallback())
                    .catch(errorCallback),
            numOfTries,
            initialTimeout,
            backoffFactor,
        ),
    );
```